R/BackendHdf5-class.R

```diff
@@ -263,7 +270,7 @@ setMethod("backendReadSpectra", "BackendHdf5", function(object, spectraData,
 #' @md
 #'
 #' @noRd
-.h5_write_spectra <- function(x, spectraData, h5file, prune = TRUE) {
+.h5_write_spectra <- function(x, spectraData, h5file, modCount, prune = TRUE) {
```

Comment:
Is `prune = TRUE` a good idea? If we allow arbitrary splitting factors in `spectrapply` (and if we are automatically and immediately writing the data back to hdf) we could remove spectra we have written in a different thread, e.g.:
```r
# 1 file with 10 spectra
spectrapply(mse, f=rep(1:2, each=5), removePeaks)
# - first thread, write spectra, remove all spectra data and write spectra 1:5 to file 1
# - second thread, write spectra, remove all spectra data and write spectra 6:10 to file 1
#  (spectra 1:5 are now lost)
```

Reply:
Writing data to hdf5 takes quite some time - that was one reason why I would not like to write every change immediately. And from a performance perspective it's much faster to remove the group within the file that contains all spectra and save all spectra in one go. Overwriting individual data sets does not work (if the dimension of the matrices differ). For now I would like to keep it that way.

Reply:
While I guess `h5delete(h5, "spectra")` is faster than `for (i in seq(along(x)) { h5delete(h5, name=spids[i]) }` the latter would IMHO be safer (and if you use again the raw internal API it should be relatively fast).

We could keep it in that way for now but we have to keep an eye on it.

Reply:
It depends on what you have in mind. For me, with the present code, it makes sense to always write all spectra of one file and never a single spectrum. And for that use case its saver to delete all content before. Imagine that you e.g. subsetted `spectraData` with `filterRt` before, now you have spectra in the hdf5 file that are no longer in `spectraData`.

In any rate, I agree that we might want to change this later if there is need.